csrc/id_model/indexing.cpp

```diff
@@ -597,33 +621,43 @@ std::pair<std::vector<ValGroup>, std::vector<Val*>> TensorIndexer::
       {contig_strides.begin(), contig_strides.end()}};
 }
 
-ValGroups TensorIndexer::getUsedLoopGroups(
-    const IndexingInfo& index_info) const {
-  ValGroups used_loop_groups;
-  for (const auto& index_id : index_info.index_ids) {
-    const ValGroups& loop_groups = index_info.loop_group_dependencies.at(
-        traversalGraph().toGroup(index_id));
-    used_loop_groups.pushBack(loop_groups);
+std::vector<ForLoop*> TensorIndexer::getUsedForLoopsOf(
```

Comment:
Previously, this was done based on the connectivity in the indexing graph. However, that doesn't guarantee the loop index of a connected loop ID is always actually used in the final index.